### PR TITLE
settings modal, local storage and darkmode toggling implemented

### PR DESCRIPTION
**Pull request by JenniferJa**

### CODE REVIEW

File: client/src/contexts/AppStateContext.tsx [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+  useState,
+} from "react";
 
 interface Props {
   children: ReactNode | ReactNode[];
 }
 
-enum speedUnits {
+export enum connectionTypes {
+  "Network",
```
Comment (ideen1) [inline client/src/contexts/AppStateContext.tsx:14]: Add a value to the Enums like: `"Network" = NETWORK`

File: client/src/contexts/AppStateContext.tsx [hunk 2]
```diff
@@ -26,22 +37,69 @@ const appStateContext = createContext<IAppStateReturn>({} as IAppStateReturn);
 
 export function AppStateContextProvider({ children }: Props) {
   const [currentAppState, setCurrentAppState] = useState<IAppState>({
-    loading: false,
+    loading: true,
     error: false,
     darkMode: false,
     speedUnits: speedUnits["km/h"],
+    connectionTypes: connectionTypes["Network"],
```
Comment (ideen1) [inline client/src/contexts/AppStateContext.tsx:44]: It's more common to access enums like this: `ConnectionTypes.NETWORK`

File: client/src/contexts/AppStateContext.tsx [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
+  useEffect,
+  useState,
+} from "react";
 
 interface Props {
   children: ReactNode | ReactNode[];
 }
 
-enum speedUnits {
+export enum connectionTypes {
```
Comment (ideen1) [inline client/src/contexts/AppStateContext.tsx:13]: Our convention is to use CapitalCase for Enum and Type Names

File: client/src/contexts/AppStateContext.tsx [hunk 4]
```diff
@@ -26,22 +37,69 @@ const appStateContext = createContext<IAppStateReturn>({} as IAppStateReturn);
...
     setCurrentAppState({
       ...currentAppState,
       darkMode: !currentAppState.darkMode,
     });
   };
 
+  const saveSettingsToLocalStorage = () => {
+    localStorage.setItem(
+      "settings",
+      JSON.stringify({
```
Comment (ideen1) [inline client/src/contexts/AppStateContext.tsx:76]: You cause just directly pass in `currentAppState` here.

File: client/src/contexts/AppStateContext.tsx [hunk 5]
```diff
@@ -26,22 +37,69 @@ const appStateContext = createContext<IAppStateReturn>({} as IAppStateReturn);
...
+  const fetchSettingsFromLocalStorage = () => {
+    const savedSettings = localStorage.getItem("settings");
+    if (savedSettings) {
+      const {
+        error: savedError,
+        darkMode: savedDarkMode,
+        speedUnits: savedSpeedUnits,
+        connectionTypes: savedConnectionTypes,
+      } = JSON.parse(savedSettings) as IAppState;
+      setCurrentAppState({
```
Comment (ideen1) [inline client/src/contexts/AppStateContext.tsx:56]: You can also just directly ready this from local storage without recreating the struct.

**Review by ideen1 — CHANGES_REQUESTED** [review]

**Review by brian-ngyn — CHANGES_REQUESTED** [review]

**Review by ideen1 — APPROVED** [review]

**Review by brian-ngyn — APPROVED** [review]

**Pull request merged**